### PR TITLE
Added round function to gles2

### LINKED ISSUE #24850: 3.1 OpenGL ES2.0 using round() causes shaders to not compile
Godot 3.1 Beta1
Windows 10
OpenGL ES 2.0 Renderer: GeForce GTX 1050/PCIe/SSE2

Compiling a shader using round() fails shader compilation, produces:
```
0(478) : error C7532: global function round requires "#version 130" or later
0(478) : error C0000: ... or #extension GL_EXT_gpu_shader4 : enable
```

To reproduce, create a basic scene with a mesh using this ShaderMaterial:
```gdscript
shader_type spatial;
render_mode unshaded;

void fragment() 
{
	ALBEDO = round( vec3(0.5, 0.0, 0.0) );
}
```

### PR DESCRIPTION
**Pull request by clayjohn**
Fixes: https://github.com/godotengine/godot/issues/24850

Implements the ``round()`` function in GLES2. Something similar can be done with a few of the other GLES3-only features found in http://docs.godotengine.org/en/latest/tutorials/misc/gles2_gles3_differences.html#shading-language

Putting this up first to see if this is the implementation we want to follow. Then I will make a PR with a few others. 

**Implementation Notes:**

Right now, I have copied the implementation from the ``mix(a, b, bool_type)`` override. The shader compiler replaces calls to ``round()`` with calls to ``roundx()``, which are defined in stdlib.glsl. This implementation works and is safe to use. However it may be unnecessary. 

An alternative would be to just provide an overloaded ``round()`` function in stdlib.glsl without changing anything in the compiler. This seems to work, but some GLES2 devices that deviate from spec allow users to use ``round()``, so I'm worried that it would crash on those devices. That being said, it is a much more elegant solution.

### CODE REVIEW

**Comment by reduz** [comment]
My only fear with adding more code to the base shaders is that versions will take longer to compile because more needs to be parsed. Maybe having some ifdef around these being used may work?

**Comment by akien-mga** [comment]
Thanks!

File: drivers/gles2/shader_compiler_gles2.cpp [hunk 1]
```diff
@@ -917,6 +928,7 @@ ShaderCompilerGLES2::ShaderCompilerGLES2() {
 	actions[VS::SHADER_CANVAS_ITEM].usage_defines["NORMAL"] = "#define NORMAL_USED\n";
 	actions[VS::SHADER_CANVAS_ITEM].usage_defines["NORMALMAP"] = "#define NORMALMAP_USED\n";
 	actions[VS::SHADER_CANVAS_ITEM].usage_defines["LIGHT"] = "#define USE_LIGHT_SHADER_CODE\n";
+	actions[VS::SHADER_CANVAS_ITEM].usage_defines["round"] = "#define ROUND_USED\n";
```
Comment (raphael10241024) [inline drivers/gles2/shader_compiler_gles2.cpp]: One question about it,  why don't add usage_defines for spatial shader here?

Comment (clayjohn) [reply drivers/gles2/shader_compiler_gles2.cpp]: Ah. Good catch. I added just for CanvasItem while testing and then forgot to add to Spatial before comitting.

Comment (Chaosus) [reply drivers/gles2/shader_compiler_gles2.cpp]: And for Particles too ?

Comment (clayjohn) [reply drivers/gles2/shader_compiler_gles2.cpp]: This is only for GLES2. So no particle shaders. ;p

**Pull request merged**